src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/Preview.js

```diff
@@ -139,14 +171,18 @@ class Preview extends React.Component<Props> {
         this.schemaDisposer = reaction(
             () => toJS(formStore.schema),
             () => {
-                previewStore.updateContext(toJS(formStore.type), toJS(formStore.data)).then(this.setContent);
+                if (formStore.type) {
+                    previewStore.updateContext(toJS(formStore.type), toJS(formStore.data)).then(this.setContent);
+                }
             }
         );
     };
 
     updatePreview = debounce((data: Object) => {
-        const {previewStore} = this;
-        previewStore.update(data).then(this.setContent);
+        if (this.props.formStore.resourceKey === this.previewStore.resourceKey) {
```

Comment:
maybe we should extract the `this.props.formStore.resourceKey === this.previewStore.resourceKey` into a computed getter to make clear what this means? eg call it `shouldUpdateFormStore`?